src/app/features/tech-record/create-batch/components/batch-vehicle-template/batch-vehicle-template.component.ts

```diff
@@ -95,40 +105,40 @@ export class BatchVehicleTemplateComponent {
     const check = this.isVehicleStatusValid;
 
     if (!this.isInvalid && check) {
-      this.technicalRecordService.editableVehicleTechRecord$
+      this.store
+        .select(selectTechRecord)
         .pipe(
           withLatestFrom(this.batchTechRecordService.batchVehicles$),
           take(1),
           map(([record, batch]) =>
             batch.map(
               v =>
                 ({
-                  ...record!,
-                  techRecord: [
-                    { ...record?.techRecord[0], statusCode: this.form.value.vehicleStatus ?? StatusCodes.PROVISIONAL } as unknown as TechRecordModel
-                  ],
+                  ...record,
+                  techRecord_statusCode: this.form.value.vehicleStatus ?? (StatusCodes.PROVISIONAL as string),
                   vin: v.vin,
                   vrms: v.vehicleType !== VehicleTypes.TRL && v.trailerIdOrVrm ? [{ vrm: v.trailerIdOrVrm, isPrimary: true }] : null,
                   trailerId: v.vehicleType === VehicleTypes.TRL && v.trailerIdOrVrm ? v.trailerIdOrVrm : null,
                   systemNumber: v.systemNumber ? v.systemNumber : null,
                   oldVehicleStatus: v.oldVehicleStatus ? v.oldVehicleStatus : null
-                } as BatchUpdateVehicleModel)
+                } as unknown as BatchUpdateVehicleModel)
             )
           )
         )
         .subscribe(vehicleList => {
           vehicleList.forEach(vehicle => {
             if (!vehicle.systemNumber) {
-              this.store.dispatch(createVehicleRecord({ vehicle }));
+              this.store.dispatch(createVehicleRecord({ vehicle: vehicle as unknown as V3TechRecordModel }));
             } else {
-              this.technicalRecordService.updateEditingTechRecord(vehicle);
-              this.store.dispatch(
-                updateTechRecords({
-                  systemNumber: vehicle.systemNumber,
-                  recordToArchiveStatus: vehicle.oldVehicleStatus ?? StatusCodes.PROVISIONAL,
-                  newStatus: vehicle.techRecord[0]?.statusCode ?? StatusCodes.CURRENT
-                })
-              );
+              this.technicalRecordService.updateEditingTechRecord(vehicle as unknown as V3TechRecordModel);
+              // this.store.dispatch(
+              //   updateTechRecords()
+              // {
+              //   systemNumber: vehicle.systemNumber,
+              //   recordToArchiveStatus: vehicle.oldVehicleStatus ?? StatusCodes.PROVISIONAL,
+              //   newStatus: vehicle.techRecord[0]?.statusCode ?? StatusCodes.CURRENT
+              // })
+              // );
```

Comment:
should this still be commented out? if not perhaps a TODO should be left so it doesnt get forgotten